tests/nn/checkpoint/test_checkpoint_activations.py

```diff
@@ -267,3 +267,38 @@ def test_deprecated_path():
     # Check if direct import works as before.
     ffn = nn.Sequential(nn.Linear(32, 128), nn.Dropout(p=0.5), nn.Linear(128, 32),)
     ffn = deprecated_checkpoint_wrapper(ffn, {})
+
+
+@skip_if_no_cuda
+def test_list_input():
+    """Test to ensure that we correctly does the checkpoint when the input is a list.
+
+    Note: Testing shows that PyTorch's torch.utils.checkpoint function does not pass this test.
+    """
+    count = 0
+
+    class Model(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.conv = nn.Linear(2, 2)
+
+        def forward(self, x):
+            nonlocal count
+            count += 1
+            y = []
```

Comment:
nit: prefer using full names to describe variables instead of single letters.

Reply:
In general I agree. Here however it is conventional to use `x` for the forward input. I think in terms of math, using `y` in this simple case is also acceptable. In bigger and more complicated function, I would not use y.